internal/resources/kustomization/spec/cluster_scope.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package spec
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	kustomizationclustermodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kustomization/cluster"
+)
+
+func ConstructSpecForClusterScope(d *schema.ResourceData) (spec *kustomizationclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceFluxcdKustomizationSpec) {
+	value, ok := d.GetOk(SpecKey)
+	if !ok {
+		return spec
+	}
+
+	data, _ := value.([]interface{})
+
+	if len(data) == 0 || data[0] == nil {
+		return spec
+	}
+
+	specData := data[0].(map[string]interface{})
```

Comment:
panic

Reply:
We have a length check for the data and a nil check for the first element right before this type assertion.

```
if len(data) == 0 || data[0] == nil {
	return spec
}
```

I have manually tested and verified the spec block multiple times. I did not encounter a panic yet.


Also, we have been following this in every other spec implementation -
[resources/policy/kind/security/spec_schema.go#L45](https://github.com/vmware/terraform-provider-tanzu-mission-control/blob/143c1e710f639065d1924cfe57d479be64720867/internal/resources/policy/kind/security/spec_schema.go#L45)